client/src/featureform/resources.py

```
@@ -1626,19 +1625,25 @@ def path(self):
+class ResourceVariant(HasNameVariant, Protocol):
     name: str
     variant: str
     server_status: ServerStatus
 
-    @staticmethod
-    def get_resource_type():
-        raise NotImplementedError
-
-    def name_variant(self):
-        return self.name, self.variant
+    def get_resource_type(self) -> ResourceType:
```

Ruff (0.8.2)

`ResourceType` may be undefined, or defined from star imports

(F405)